src/libcommonserver/utility/utility.cpp

```diff
@@ -390,10 +390,27 @@ bool Utility::endsWithInsensitive(const std::string &str, const std::string &suf
                       [](char c1, char c2) { return std::tolower(c1, std::locale()) == std::tolower(c2, std::locale()); });
 }
 
-bool Utility::isEqualInsensitive(const std::string &strA, const std::string &strB) {
-    return strA.size() == strB.size() && std::equal(strA.begin(), strA.end(), strB.begin(), strB.end(), [](char a, char b) {
-               return std::tolower(a, std::locale()) == std::tolower(b, std::locale());
-           });
+bool Utility::isEqualUpToCaseAndEnc(const SyncPath &a, const SyncPath &b, bool &isEqual) {
+    SyncPath normalizedA;
+    if (!Utility::normalizedSyncPath(a, normalizedA)) {
+        LOGW_WARN(_logger, L"Error in Utility::normalizedSyncPath: " << Utility::formatSyncPath(a));
+        return false;
+    }
+
+    SyncPath normalizedB;
+    if (!Utility::normalizedSyncPath(b, normalizedB)) {
+        LOGW_WARN(_logger, L"Error in Utility::normalizedSyncPath: " << Utility::formatSyncPath(b));
+        return false;
+    }
+
+    const SyncName normalizedAStr{normalizedA.native()};
+    const SyncName normalizedBStr{normalizedB.native()};
+
+    isEqual = normalizedAStr.size() == normalizedAStr.size() &&
```

Comment:
```suggestion
    isEqual = normalizedAStr.size() == normalizedBStr.size() &&
```

Spotted by SonarCloud!